UnauthorizedAccessException: Access to the path 'C:\Documents and Settings' is denied.

I have a problem with generating swagger doc using command line tool
I have NetCore 2.2 webapi
I've installed NSwag.MsBuild package and added following to the csproj:

```
<Target Name="NSwag" AfterTargets="Build">
<Copy SourceFiles="@(Reference)" DestinationFolder="$(OutDir)References" />
<Exec Command="$(NSwagExe_Core22) aspnetcore2swagger /assembly:/CloudFabric.ConfigurationServer.WebApi.dll /output:swagger.json /runtime:NetCore22" />
<RemoveDir Directories="$(OutDir)References" />
<Exec Command="$(NSwagExe_Core22) swagger2tsclient /input:swagger.json /output:Scripts/ConfigurationServer.ts" />
<Exec Command="$(NSwagExe_Core22) swagger2csclient /input:swagger.json /output:CSClient/ConfigurationServer.cs" />
</Target>
```

When running I get the error from aspnetcore2swagger :

```
NSwag command line tool for .NET Core NetCore22, toolchain v12.2.5.0 (NJsonSchema v9.13.37.0 (Newtonsoft.Json v11.0.0.0))
>Visit http://NSwag.org for more information.
>NSwag bin directory: C:\Users\Jakub Konecki.nuget\packages\nswag.msbuild\12.2.5\tools\NetCore22
>System.UnauthorizedAccessException: Access to the path 'C:\Documents and Settings' is denied.
> at System.IO.Enumeration.FileSystemEnumerator1.CreateRelativeDirectoryHandle(ReadOnlySpan1 relativePath, String fullPath)
> at System.IO.Enumeration.FileSystemEnumerator1.MoveNext() 3> at System.Collections.Generic.LargeArrayBuilder1.AddRange(IEnumerable1 items) 3> at System.Collections.Generic.EnumerableHelpers.ToArray[T](IEnumerable1 source)
> at System.Linq.Enumerable.ToArrayTSource
> at NSwag.AssemblyLoader.AssemblyLoader.RegisterReferencePaths(IEnumerable1 referencePaths) in C:\projects\nswag\src\NSwag.AssemblyLoader\AssemblyLoader.cs:line 87 3> at NSwag.Commands.IsolatedCommandBase1.IsolatedCommandAssemblyLoader1.Run(String commandType, String commandData, String[] assemblyPaths, String[] referencePaths) in C:\projects\nswag\src\NSwag.Commands\Commands\IsolatedCommandBase.cs:line 72 3> at NSwag.Commands.IsolatedCommandBase1.<>cDisplayClass17_0.<RunIsolatedAsync>b0() in C:\projects\nswag\src\NSwag.Commands\Commands\IsolatedCommandBase.cs:line 60
> at System.Threading.Tasks.Task1.InnerInvoke() 3> at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state) 3>--- End of stack trace from previous location where exception was thrown --- 3> at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot) 3>--- End of stack trace from previous location where exception was thrown --- > at NSwag.Commands.IsolatedCommandBase1.RunIsolatedAsync(String configurationFile)
> at NSwag.Commands.IsolatedSwaggerOutputCommandBase`1.RunAsync(CommandLineProcessor processor, IConsoleHost host) in C:\projects\nswag\src\NSwag.Commands\Commands\IsolatedSwaggerOutputCommandBase.cs:line 47
> at NSwag.Commands.SwaggerGeneration.AspNetCore.AspNetCoreToSwaggerCommand.RunAsync(CommandLineProcessor processor, IConsoleHost host) in C:\projects\nswag\src\NSwag.Commands\Commands\SwaggerGeneration\AspNetCore\AspNetCoreToSwaggerCommand.cs:line 88
> at NConsole.CommandLineProcessor.ProcessSingleAsync(String[] args, Object input)
> at NConsole.CommandLineProcessor.ProcessAsync(String[] args, Object input)
> at NConsole.CommandLineProcessor.Process(String[] args, Object input)
> at NSwag.Commands.NSwagCommandProcessor.Process(String[] args) in C:\projects\nswag\src\NSwag.Commands\NSwagCommandProcessor.cs:line 54
>C:\Projects\TechFabric\CloudFabric.ConfigurationServer\CloudFabric.ConfigurationServer.WebApi\CloudFabric.ConfigurationServer.WebApi.csproj(25,5): error MSB3073: The command "dotnet "C:\Users\Jakub Konecki.nuget\packages\nswag.msbuild\12.2.5\build../tools/NetCore22/dotnet-nswag.dll" aspnetcore2swagger /assembly:/CloudFabric.ConfigurationServer.WebApi.dll /referencePaths:bin\Debug\netcoreapp2.2\References /output:swagger.json /runtime:NetCore22" exited with code -1.
```

I run Win 10 and don't have 'C:\Documents and Settings' folder (just 'C:\Users')

--- a/src/NSwag.AssemblyLoader/AssemblyLoader.cs
+++ b/src/NSwag.AssemblyLoader/AssemblyLoader.cs
@@ -197,7 +197,15 @@ namespace NSwag.AssemblyLoader
         private string[] GetAllDirectories(string rootDirectory)
         {
             rootDirectory = Environment.ExpandEnvironmentVariables(rootDirectory);
-            return Directory.GetDirectories(rootDirectory, "*", SearchOption.AllDirectories);
+
+            try
+            {
+                return Directory.GetDirectories(rootDirectory, "*", SearchOption.AllDirectories);
+            }
+            catch // https://github.com/RicoSuter/NSwag/issues/2177
+            {
+                return Array.Empty<string>();
+            }
         }
     }
 }

